File: spring-boot-rest-demo/src/main/java/com/github/jccode/springbootrestdemo/controller/advice/GlobalControllerExceptionHandler.java
package com.github.jccode.springbootrestdemo.controller.advice;

import com.github.jccode.springbootrestdemo.common.data.Failed;
import com.github.jccode.springbootrestdemo.exception.RestException;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseBody;

import javax.servlet.http.HttpServletRequest;

@ControllerAdvice
public class GlobalControllerExceptionHandler {

    @ResponseBody
    @ExceptionHandler(RestException.class)
    public Failed restException(Exception e, HttpServletRequest request) {
        return new Failed(e.getMessage());
    }


}

File: spring-boot-rest-demo/src/main/scala/com/github/jccode/springbootrestdemo/controller/advice/GlobalControllerExceptionHandler.scala
package com.github.jccode.springbootrestdemo.controller.advice

import javax.servlet.http.HttpServletRequest

import com.github.jccode.springbootrestdemo.common.Failed
import com.github.jccode.springbootrestdemo.exception.RestException
import org.springframework.web.bind.annotation.{ControllerAdvice, ExceptionHandler, ResponseBody}

@ControllerAdvice
class GlobalControllerExceptionHandler {

  @ResponseBody
  @ExceptionHandler(Array(classOf[RestException]))
  def restException(e: Exception, request: HttpServletRequest): Failed =
    new Failed(e.getMessage)

}
